sdk/metric/exemplar/reservoir_test.go

```diff
@@ -144,3 +146,66 @@ func ReservoirTest[N int64 | float64](f factory) func(*testing.T) {
 		})
 	}
 }
+
+func reservoirConcurrentSafeTest[N int64 | float64](f factory) func(*testing.T) {
+	return func(t *testing.T) {
+		t.Helper()
+		rp, n := f(1)
+		if n < 1 {
+			t.Skip("skipping, reservoir capacity less than 1:", n)
+		}
+		r := rp(*attribute.EmptySet())
+
+		var wg sync.WaitGroup
+
+		// Call Offer concurrently with another Offer, and with Collect.
+		for i := range 2 {
+			wg.Add(1)
+			go func() {
+				ctx, ts, val, attrs := generateOfferInputs[N](t, i+1)
+				r.Offer(ctx, ts, val, attrs)
+				wg.Done()
+			}()
```

Comment:
nit: use explicit parameter passing to improve clarity for older Go versions that the loop var is captured safely.

```suggestion
			go func(interation int) {
				ctx, ts, val, attrs := generateOfferInputs[N](t, iteration+1)
				r.Offer(ctx, ts, val, attrs)
				wg.Done()
			}(i)
```